engine/vcs/bitbucket/client_branch.go

```diff
@@ -69,11 +69,29 @@ func (b *bitbucketClient) Branch(fullname string, filter string) (*sdk.VCSBranch
 		return nil, sdk.ErrNotFound
 	}
 
-	branch := &sdk.VCSBranch{
-		ID:           branches.Values[0].ID,
-		DisplayID:    branches.Values[0].DisplayID,
-		LatestCommit: branches.Values[0].LatestHash,
-		Default:      branches.Values[0].IsDefault,
+	found := false
+	var branch sdk.VCSBranch
+	for _, b := range branches.Values {
+		if b.DisplayID == filter {
+			found = true
+			branch = sdk.VCSBranch{
+				ID:           b.ID,
+				DisplayID:    b.DisplayID,
+				LatestCommit: b.LatestHash,
+				Default:      b.IsDefault,
+			}
+			break
+		}
+	}
+
+	if !found {
+		branch = sdk.VCSBranch{
```

Comment:
so if it's not found, you will return a wrong branch?
Why not fail instead?
Failure is safer IMHO
I dont want to build a branch that looks like the one I am expecting. If I did something wrong, I just want the system to tell me fuck you

Reply:
MIssing something yes, but not sure where, api side or vcs µservice.